lib/heuristics/concerns/scheduling_end_phase.rb

```diff
@@ -337,14 +338,16 @@ def reaffect_in_non_empty_route(still_removed)
   end
 
   def collect_empty_routes
-    empty_routes = @candidate_routes.collect{ |vehicle, data|
-      next unless @candidate_routes[vehicle].any?{ |_day, day_data| day_data[:current_route].empty? }
+    empty_routes = @candidate_routes.collect{ |vehicle_id, all_routes|
+      next unless @candidate_routes[vehicle_id].any?{ |_day, route_data| route_data[:stops].empty? }
 
-      data.collect{ |day, day_data|
-        if day_data[:current_route].empty?
-          [vehicle,
+      all_routes.collect{ |day, route_data|
+        if route_data[:stops].empty?
+          [vehicle_id,
            day,
-           stores: [day_data[:start_point_id], day_data[:end_point_id]], time_range: day_data[:tw_end] - day_data[:tw_start]]
+           stores: [
+             route_data[:start_point_id], route_data[:end_point_id]], time_range: route_data[:tw_end] - route_data[:tw_start]
+            ]
```

Comment:
This formatting looks a bit incorrect :thinking: or it is github making it look incorrect... 

What I meant was something like the following (if that is what we want to collect)
```ruby
[
  vehicle_id,
  day,
  {
    stores: [route_data[:start_point_id], route_data[:end_point_id]], 
    time_range: route_data[:tw_end] - route_data[:tw_start]
  }
]
``` 
it is easier to see the collected items